File: CS.DotNetCore.LoadTest/src/CS.DotNetCore.LoadTest.WebApp/Business/Identity.cs
namespace CS.DotNetCore.LoadTest.WebApp.Business
{
    using Config;
    using Newtonsoft.Json;
    using System.Security.Cryptography;

    public class Identity
    {
        [JsonProperty]
        internal byte[] Salt { get; private set; }

        [JsonProperty]
        internal byte[] Password { get; private set; }

        [JsonProperty]
        internal string IdentityName { get; private set; }

        internal Identity(string identityName)
        {
            IdentityName = identityName;
        }

        internal void SetPassword(string password, ILoadTestConfig config)
        {
            var saltBytes = config.Encoding.GetBytes(config.Security.PasswordSalt);

            using (var aes = Aes.Create())
            {
                using (var cipher = aes.CreateEncryptor(config.Security.PasswordSaltKey, config.Security.PasswordSaltIV))
                {
                    Salt = cipher.TransformFinalBlock(saltBytes, 0, saltBytes.Length);
                }
            }

            var saltedPassword = config.Security.PasswordSalt + password;
            var saltedPasswordBytes = config.Encoding.GetBytes(saltedPassword);

            using (var sha = SHA256.Create())
            {
                Password = sha.ComputeHash(saltedPasswordBytes);
            }
        }
    }
}

File: DotNetCorePracticalView/CS.NodeJS.LoadTest.WebApp/webapp/business/identity.js
let crypto = require('crypto');

module.exports = {

  create: (identityName) => {

    let self = {

      identityName: identityName,
      salt: null,
      password: null
    };

    self.setPassword = (password, config) => {

      const outputEncoding = 'base64',
        cipherAlg = 'aes256',
        hashAlg = 'sha256';

      let saltIV = Buffer.from(config.security.passwordSaltIV, outputEncoding);
      let saltKey = Buffer.from(config.security.passwordSaltKey, outputEncoding);

      let aesCipher = crypto.createCipheriv(cipherAlg, saltKey, saltIV);
      aesCipher.update(config.security.passwordSalt, config.encoding);
      self.salt = aesCipher.final(outputEncoding);

      let shaAlg = crypto.createHash(hashAlg);
      shaAlg.update(config.security.passwordSalt + password, config.encoding);
      self.password = shaAlg.digest(outputEncoding);
    }

    return self;
  }
}